File: ios/HelloWorldObjc/HelloWorldObjc/ViewController.m

#import "ViewController.h"
#import <DynamsoftCameraEnhancer/DynamsoftCameraEnhancer.h>

@interface ViewController ()<DCEFrameListener>

@property(nonatomic, strong) DynamsoftCameraEnhancer *dce;
@property(nonatomic, strong) DCECameraView *dceView;

@end

@implementation ViewController{
    UIButton *photoButton;
    UIImageView* imageView;
    bool isview;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    [self configurationDCE];
    [self configurationUI];
}

- (void)configurationDCE{
    _dceView = [DCECameraView cameraWithFrame:self.view.bounds];
    [self.view addSubview:_dceView];
    _dce = [[DynamsoftCameraEnhancer alloc] initWithView:_dceView];
    [_dce open];
    [_dce addListener:self];
    [_dce setFrameRate:30];
}

- (void)configurationUI{
    CGFloat w = [[UIScreen mainScreen] bounds].size.width;
    CGFloat h = [[UIScreen mainScreen] bounds].size.height;
    CGFloat SafeAreaBottomHeight = [[UIApplication sharedApplication] statusBarFrame].size.height > 20 ? 34 : 0;
    photoButton = [[UIButton alloc] initWithFrame:CGRectMake(w / 2 - 60, h - 170 - SafeAreaBottomHeight, 120, 120)];
    photoButton.adjustsImageWhenDisabled = NO;
    [photoButton setImage:[UIImage imageNamed:@"icon_capture"] forState:UIControlStateNormal];
    self->imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, w, h)];
    [photoButton addTarget:self action:@selector(takePictures) forControlEvents:UIControlEventTouchUpInside];
    dispatch_async(dispatch_get_main_queue(), ^{
        [self.view addSubview:self->photoButton];
    });
}

- (void)takePictures{
    isview = true;
}

- (void)frameOutPutCallback:(nonnull DCEFrame *)frame timeStamp:(NSTimeInterval)timeStamp {
    if (isview) {
        isview = false;
        dispatch_async(dispatch_get_main_queue(), ^{
            [self->photoButton setEnabled:false];
            UIImage *image = [[UIImage alloc] initWithCGImage: frame.toUIImage.CGImage
                                               scale: 1.0
                                         orientation: UIImageOrientationRight];
            [self->imageView setImage:image];
            [self.view addSubview:self->imageView];
            [self addBack];
        });
    }
}

- (void)addBack{
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemReply target:self action:@selector(BackToHome)];
}

- (void)BackToHome{
    [imageView removeFromSuperview];
    self.navigationItem.leftBarButtonItem = nil;
    [photoButton setEnabled:true];
}

@end

File: ios/HelloWorldSwift/HelloWorldSwift/ViewController.swift
//
//  ViewController.swift
//
//

import UIKit

class ViewController: UIViewController,DCEFrameListener{
    
    var photoButton:UIButton! = UIButton()
    var imageView:UIImageView!
    var isview:Bool = false
    
    
    var dce:DynamsoftCameraEnhancer! = nil
    var dceView:DCECameraView! = nil
    override func viewDidLoad() {
        super.viewDidLoad()
        configurationDCE()
        configurationUI()
    }
    
    func configurationDCE() {
        dceView = DCECameraView.init(frame: self.view.bounds)
        self.view.addSubview(dceView)
        dce = DynamsoftCameraEnhancer.init(view: dceView)
        dce.open()
        dce.addListener(self)
    }
    
    func configurationUI() {
        let w = UIScreen.main.bounds.size.width
        let h = UIScreen.main.bounds.size.height
        let safeAreaBottomHeight:CGFloat = UIApplication.shared.statusBarFrame.size.height > 20 ? 34 : 0
        photoButton = UIButton(frame: CGRect(x:w / 2 - 60, y: h - 170 - safeAreaBottomHeight, width: 120, height: 120))
        photoButton.adjustsImageWhenDisabled = false
        photoButton.setImage(UIImage(named: "icon_capture"), for: .normal)
        self.imageView = UIImageView(frame: CGRect(x: 0, y: 0, width: w, height: h))
        photoButton.addTarget(self, action: #selector(takePictures), for: .touchUpInside)
        DispatchQueue.main.async {
            self.view.addSubview(self.photoButton)
        }
    }
    
    @objc func takePictures() {
        isview  = true
    }
    
    func frameOutPutCallback(_ frame: DCEFrame, timeStamp: TimeInterval) {
        if isview {
            isview = false
            DispatchQueue.main.async {
                self.photoButton?.isEnabled = false
                var image:UIImage!
                image = frame.toUIImage()
                image = UIImage.init(cgImage: image.cgImage!, scale: 1.0, orientation: UIImageOrientation.right)
                self.imageView.image = image
                self.view.addSubview(self.imageView)
                self.addBack()
            }
        }
    }
    
    func addBack(){
        self.navigationItem.leftBarButtonItem = UIBarButtonItem.init(barButtonSystemItem: .reply, target: self, action: #selector(backToHome))
    }
    
    @objc func backToHome(){
        self.imageView.removeFromSuperview()
        self.photoButton?.isEnabled = true
        self.navigationItem.leftBarButtonItem = nil
    }

}

